plugins/faustwp/includes/replacement/callbacks.php

```diff
@@ -50,16 +55,42 @@ function content_replacement( ?string $content ) {
 	$relative_upload_url = faustwp_get_relative_upload_url( $wp_site_urls, wp_upload_dir()['baseurl'] );
 	$wp_media_urls       = faustwp_get_wp_media_urls( $wp_site_urls, $relative_upload_url );
 	$frontend_uri        = (string) faustwp_get_setting( 'frontend_uri' ) ?? '/';
-	if ( $replace_content_urls && $replace_media_urls ) {
-		return str_replace( $wp_site_urls, $frontend_uri, $content );
+
+	/* If "Enable Post and Category URL" is enabled, use front-end URL for internal URLs, but not for media links */
+
+	if ( $replace_content_urls ) {
+
+		// Look for href links.
+		preg_match_all( '#href="([^"]+)"#i', $content, $href_links );
+
+		foreach ( $href_links[1] as $i => $url ) {
+			// skip media links.
+			$is_media = array_filter( $wp_media_urls, fn( $media ) => strpos( $url, $media ) === 0 );
+			if ( $is_media ) {
+				continue;
+			}
+
+			$is_wp_url = array_filter( $wp_site_urls, fn( $base ) => strpos( $url, $base ) === 0 );
+			if ( ! $is_wp_url ) {
+				continue;
+			}
+
+			// get relative link.
+			$relative = str_replace( reset( $is_wp_url ), '', $url );
+			$updated  = 'href="' . $frontend_uri . $relative . '"';
+
+			$original = $href_links[0][ $i ];
```

Comment:
@alexwpengine 

I think we should check that key exists before doing any string replace